user_api/models.py

```diff
@@ -1,15 +1,60 @@
 from django.db import models
+from django.contrib.auth.models import (
+    BaseUserManager, AbstractBaseUser
+)
 
-# Create your models here.
-
-class User(models.Model):
-    name = models.CharField(max_length=50)
-    email=models.EmailField()
-    password=models.CharField(max_length=30)
+class User(AbstractBaseUser):
+    email = models.EmailField(
+        verbose_name='email address',
+        max_length=255,
+        unique=True,
+    )
     city=models.CharField(max_length=50)
     state=models.CharField(max_length=50)
     zipcode=models.IntegerField()
     balance=models.IntegerField()
     phone=models.BigIntegerField()
     created_at=models.DateTimeField(auto_now_add=True)
-    updated_at=models.DateTimeField(auto_now_add=True)
\ No newline at end of file
+    updated_at=models.DateTimeField(auto_now_add=True)
+
+    is_active = models.BooleanField(default=True)
+    staff = models.BooleanField(default=False) # a admin user; non super-user
+    admin = models.BooleanField(default=False) # a superuser
+
+    # notice the absence of a "Password field", that is built in.
+
+    USERNAME_FIELD = 'email'
+    REQUIRED_FIELDS = [] # Email & Password are required by default.
+
+    def get_full_name(self):
+        # The user is identified by their email address
+        return self.email
+
+    def get_short_name(self):
+        # The user is identified by their email address
+        return self.email
+
+    def __str__(self):
+        return self.email
+
+    def has_perm(self, perm, obj=None):
+        "Does the user have a specific permission?"
+        # Simplest possible answer: Yes, always
+        return True
+
+    def has_module_perms(self, app_label):
+        "Does the user have permissions to view the app `app_label`?"
+        # Simplest possible answer: Yes, always
+        return True
+
+    @property
+    def is_staff(self):
+        "Is the user a member of staff?"
+        return self.staff
+
+    @property
+    def is_admin(self):
+        "Is the user a admin member?"
+        return self.admin
```

Comment:
Why these functions are made ? Don't just copy paste the things. Check what is the use-case of these functions and why are they required

Reply:
Yes, I removed them.